src/components/scenes/TransactionDetailsScene.tsx

```diff
@@ -14,142 +14,122 @@ import { lstrings } from '../../locales/strings'
-    const txActionInfo = getTxActionDisplayInfo(edgeTransaction, wallet, tokenId)
-    const txActionSplitCat = txActionInfo?.splitCategory
-    const txActionNotes = txActionInfo?.notes
-    const txActionDir = txActionInfo?.direction
+  // Choose a default category based on metadata or the txAction
+  const txActionInfo = getTxActionDisplayInfo(transaction, wallet, tokenId)
+  const txActionSplitCat = txActionInfo?.splitCategory
+  const txActionNotes = txActionInfo?.notes
+  const txActionDir = txActionInfo?.direction
 
-    // Determine direction from EdgeTransaction nativeAmount if not specified in
-    // txActionInfo
-    const direction = txActionDir ?? isSentTransaction ? 'send' : 'receive'
+  // Determine direction from transaction nativeAmount if not specified in
+  // txActionInfo
+  const direction = txActionDir ?? isSentTransaction ? 'send' : 'receive'
 
-    const splitCat =
-      metadata?.category != null || txActionSplitCat == null
-        ? splitCategory(
-            metadata?.category,
-            // Pick the right default:
-            direction === 'receive' ? 'income' : 'expense'
-          )
-        : txActionSplitCat
+  const splitCat =
+    metadata?.category != null || txActionSplitCat == null
+      ? splitCategory(
+          metadata?.category,
+          // Pick the right default:
+          direction === 'receive' ? 'income' : 'expense'
+        )
+      : txActionSplitCat
 
-    const category = joinCategory(splitCat)
+  const category = joinCategory(splitCat)
 
-    const notes = metadata?.notes == null ? txActionNotes : metadata.notes
+  const notes = metadata?.notes == null ? txActionNotes : metadata.notes
 
-    this.state = {
-      acceleratedTx: null,
-      bizId: 0,
-      category,
-      name: metadata?.name ?? '',
-      direction,
-      notes: notes ?? ''
-    }
-  }
+  const [localMetadata, setLocalMetadata] = React.useState<EdgeMetadata>({
+    bizId: 0,
+    category,
+    name: metadata?.name ?? '',
+    notes: notes ?? ''
+  })
+  const [acceleratedTx, setAcceleratedTx] = React.useState<null | EdgeTransaction>(null)
 
-  async componentDidMount() {
-    const { route } = this.props
-    const { edgeTransaction } = route.params
+  const { name = '' } = localMetadata
 
+  React.useEffect(() => {
     // Try accelerating transaction to check if transaction can be accelerated
-    this.makeAcceleratedTx(edgeTransaction)
+    makeAcceleratedTx(transaction)
       .then(acceleratedTx => {
-        this.setState({ acceleratedTx })
+        setAcceleratedTx(acceleratedTx)
       })
       .catch(_err => {})
-  }
-
-  async makeAcceleratedTx(transaction: EdgeTransaction): Promise<EdgeTransaction | null> {
-    const { wallet } = this.props
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
 
+  const makeAcceleratedTx = async (transaction: EdgeTransaction): Promise<EdgeTransaction | null> => {
     return await wallet.accelerate(transaction)
   }
```

Comment:
Can this function body just move into the `useEffect` above? It seems kinda pointless now.

Reply:
Yeah we still need to make a slightly different call in the accelerate modal so I'll just write out the `accelerate` call directly